ui/src/app/views/workspace/component.ts

```diff
@@ -191,10 +196,43 @@ export class WorkspaceComponent implements OnInit {
   }
 
   openCluster(): void {
-    const url = 'https://leonardo.dsde-dev.broadinstitute.org/notebooks/'
+    // TODO (blrubenstein): Make this configurable by environment
+    const leoBaseUrl = 'https://leonardo.dsde-dev.broadinstitute.org';
+    const leoNotebookUrl = leoBaseUrl + '/notebooks/'
         + this.cluster.clusterNamespace + '/'
         + this.cluster.clusterName;
-    window.location.href = url;
+    const leoSetCookieUrl = leoNotebookUrl + '/setCookie';
+
+    const headers = new Headers();
+    headers.append('Authorization', 'Bearer ' + this.signInService.currentAccessToken);
+    this.http.get(leoSetCookieUrl, {
+      headers: headers,
+      withCredentials: true
+    }).subscribe(() => {
+      const notebook = window.open(leoNotebookUrl, '_blank');
+      this.clusterPulled = false;
+
+      if (!this.listenerAdded) {
+        window.addEventListener('message', (e) => {
```

Comment:
Do you need to also remove this when done? Would it make sense to add and remove this on onInit and onDestroy, respectively?